e2e/pod.go

```diff
@@ -406,3 +406,22 @@ func calculateSHA512sum(f *framework.Framework, app *v1.Pod, filePath string, op
 
 	return checkSum, nil
 }
+
+// getKernelVersionFromContainer gets the kernel version from the specified container.
+func getKernelVersionFromContainer(f *framework.Framework, ns, dsn, cn string) (string, error) {
```

Comment:
sorry, I did not convey what I meant. This function is not generic (pod, deployment, etc)  it is specific to the daemonset pod the function name should be `getKernelVersionFromDaemonset`.

Not if we have kubernetes clusters with multiple nodes (with different kernel versions) things might not work as expected. 

This can be enhanced as follows (maybe as a follow-up patch)

* Check all the nodes kernel versions
* If any of the node is having expected kernel version
* schedule the application pod to the specific node

Reply:
Got it, I can rename it to getKernelVersionFromDaemonset.

Yeah, right! we can add such intelligence, for now, let's switch the timeout based on the kernel versions.
In the future, we can improve it (I will raise an issue to track your suggestions)

Thanks Madhu.

Reply:
Here: https://github.com/ceph/ceph-csi/issues/2432